### PR TITLE
Drop Python 2 support

### PR DESCRIPTION
**Pull request by ocaisa**
Needed to be compatible with `distributed/2.x`

### CODE REVIEW

File: dask_jobqueue/core.py [hunk 1]
```diff
@@ -344,6 +344,11 @@ def worker_spec(self):
             "memory": self.worker_process_memory,
         }
 
+    @property
```
Comment (willirath) [inline dask_jobqueue/core.py]: Don't include this here, but leave it in #282.

File: dask_jobqueue/tests/test_htcondor.py [hunk 2]
```diff
@@ -71,7 +71,7 @@ def test_basic(loop):
             workers = list(client.scheduler_info()["workers"].values())
             w = workers[0]
             assert w["memory_limit"] == 1e8
-            assert w["ncores"] == 1
+            assert w["nthreads"] == 1
```
Comment (willirath) [inline dask_jobqueue/tests/test_htcondor.py]: Don't include this here, but leave it in #282.

File: dask_jobqueue/tests/test_lsf.py [hunk 3]
```diff
@@ -131,7 +131,7 @@ def test_basic(loop):
             workers = list(client.scheduler_info()["workers"].values())
             w = workers[0]
             assert w["memory_limit"] == 2e9
-            assert w["ncores"] == 2
+            assert w["nthreads"] == 2
```
Comment (willirath) [inline dask_jobqueue/tests/test_lsf.py]: Don't include this here, but leave it in #282.

File: dask_jobqueue/tests/test_pbs.py [hunk 4]
```diff
@@ -128,7 +128,7 @@ def test_basic(loop):
             workers = list(client.scheduler_info()["workers"].values())
             w = workers[0]
             assert w["memory_limit"] == 2e9
-            assert w["ncores"] == 2
+            assert w["nthreads"] == 2
```
Comment (willirath) [inline dask_jobqueue/tests/test_pbs.py]: Don't include this here, but leave it in #282.

File: dask_jobqueue/tests/test_pbs.py [hunk 5]
```diff
@@ -167,7 +167,7 @@ def test_scale_cores_memory(loop):
...
```
Comment (willirath) [inline dask_jobqueue/tests/test_pbs.py]: Don't include this here, but leave it in #282.

File: dask_jobqueue/tests/test_pbs.py [hunk 6]
```diff
@@ -332,7 +332,7 @@ def test_scale_grouped(loop):
             workers = list(client.scheduler_info()["workers"].values())
             w = workers[0]
             assert w["memory_limit"] == 1e9
-            assert w["ncores"] == 1
+            assert w["nthreads"] == 1
```
Comment (willirath) [inline dask_jobqueue/tests/test_pbs.py]: Don't include this here, but leave it in #282.

File: dask_jobqueue/tests/test_sge.py [hunk 7]
```diff
@@ -32,7 +32,7 @@ def test_basic(loop):
             workers = list(client.scheduler_info()["workers"].values())
             w = workers[0]
             assert w["memory_limit"] == 2e9 / 4
-            assert w["ncores"] == 2
+            assert w["nthreads"] == 2
```
Comment (willirath) [inline dask_jobqueue/tests/test_sge.py]: Don't include this here, but leave it in #282.

File: dask_jobqueue/tests/test_slurm.py [hunk 8]
```diff
@@ -136,7 +136,7 @@ def test_basic(loop):
             workers = list(client.scheduler_info()["workers"].values())
             w = workers[0]
             assert w["memory_limit"] == 2e9
-            assert w["ncores"] == 2
+            assert w["nthreads"] == 2
```
Comment (willirath) [inline dask_jobqueue/tests/test_slurm.py]: Don't include this here, but leave it in #282.

File: dask_jobqueue/core.py (hunk 1 above)
Comment (ocaisa) [reply dask_jobqueue/core.py]: This is required for the tests to pass, it's related to the name change upstream.

**Comment by ocaisa** [comment]
Rather than clean this up I will just open a new PR

**Pull request closed without merging**